tests/FixtureBuilder/ExpressionLanguage/Parser/TokenParser/Chainable/StringArrayTokenParserTest.php

```diff
@@ -71,7 +71,10 @@ public function testThrowsAnErrorIfCouldNotParseToken(): void
             $parser = new StringArrayTokenParser(new FakeParser());
 
             $parser->parse($token);
-            static::fail('Expected exception to be thrown.');
+
+            if (PHP_VERSION_ID < 80000) {
+                static::fail('Expected exception to be thrown.');
```

Comment:
why is it no longer the case here in PHP8?

Reply:
Because PHP8.0 function substr return only string -therefore, the parameter error typehint will never fall.